src/mtcp/mtcp_restart.c

```diff
@@ -463,11 +463,25 @@ static void restart_slow_path()
   restorememoryareas(&rinfo);
 }
 
-static void mtcp_simulateread(int fd)
+static void mtcp_simulateread(int fd, MtcpHeader *mtcpHdr)
 {
   int mtcp_sys_errno;
+
+  // Print miscellaneous information:
+  char buf[MTCP_SIGNATURE_LEN+1];
+  mtcp_memcpy(buf, mtcpHdr->signature, MTCP_SIGNATURE_LEN);
+  buf[MTCP_SIGNATURE_LEN] = '\0';
+  mtcp_printf("\nMTCP: %s", buf);
```

Comment:
Here it is okay to use `mtcp_printf` because we are printing a lot of raw data. Can you also add a MTCP_PRINTF at the top of this function that would act as a header for this raw data?